In [7]:
# importing necessary library
import pandas as pd
import numpy as np

In [8]:
# defining dataframe
data = pd.read_csv('BreastCancerPrediction.csv')
BreastCancerPrediction = pd.DataFrame(data)

In [9]:
#the first 5 rows of the data
BreastCancerPrediction.head()

Country  Nipple discharge  Change of breast shape  Thickning  colourishness  \
0  Africa                38                    72.0       35.0            0.0   
1  Africa                65                    66.0       29.0            0.0   
2  Africa               113                    64.0        0.0            0.0   
3  Africa                49                    66.0       23.0           94.0   
4  Africa               117                    40.0       35.0          168.0   

   New lump inunderarm  BreastCancerLikelihood  AgeGroup  Results  
0                 33.6                   0.627      20.0        1  
1                 26.6                   0.351      41.0        0  
2                 23.3                   0.672      52.0        1  
3                 28.1                   0.167      50.0        0  
4                 43.1                   2.288      66.0        1

In [10]:
# showing  data information
BreastCancerPrediction.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 529 entries, 0 to 528
Data columns (total 9 columns):
Country                   529 non-null object
Nipple discharge          529 non-null int64
Change of breast shape    528 non-null float64
Thickning                 527 non-null float64
colourishness             528 non-null float64
New lump inunderarm       528 non-null float64
BreastCancerLikelihood    528 non-null float64
AgeGroup                  527 non-null float64
Results                   529 non-null int64
dtypes: float64(6), int64(2), object(1)
memory usage: 37.3+ KB


In [11]:
# data description
BreastCancerPrediction.describe

<bound method NDFrame.describe of         Country  Nipple discharge  Change of breast shape  Thickning  \
0        Africa                38                    72.0       35.0   
1        Africa                65                    66.0       29.0   
2        Africa               113                    64.0        0.0   
3        Africa                49                    66.0       23.0   
4        Africa               117                    40.0       35.0   
..          ...               ...                     ...        ...   
524  Artic-Area                85                    48.0        0.0   
525  Artic-Area               112                    50.0       22.0   
526  Artic-Area                47                    84.0       49.0   
527  Artic-Area                37                    73.0       45.0   
528  Artic-Area               217                    50.0       44.0   

     colourishness  New lump inunderarm  BreastCancerLikelihood  AgeGroup  \
0              0.0      

In [12]:
# preprocessing and cleaning the data as appropriate


In [13]:
# finding missing values in the dataset
BreastCancerPrediction.isna().sum()

Country                   0
Nipple discharge          0
Change of breast shape    1
Thickning                 2
colourishness             1
New lump inunderarm       1
BreastCancerLikelihood    1
AgeGroup                  2
Results                   0
dtype: int64

In [14]:
# change categorical data to quantitative data
from sklearn.preprocessing import OneHotEncoder
hotencoder = OneHotEncoder()

In [15]:
country = hotencoder.fit_transform(BreastCancerPrediction[['Country']]).toarray()
country_frame = pd.DataFrame(country)

# creating a new dataframe to include the transformed data
BreastCancerPrediction_clean = pd.concat([BreastCancerPrediction,country_frame],axis=1)

In [16]:
BreastCancerPrediction_clean

Country  Nipple discharge  Change of breast shape  Thickning  \
0        Africa                38                    72.0       35.0   
1        Africa                65                    66.0       29.0   
2        Africa               113                    64.0        0.0   
3        Africa                49                    66.0       23.0   
4        Africa               117                    40.0       35.0   
..          ...               ...                     ...        ...   
524  Artic-Area                85                    48.0        0.0   
525  Artic-Area               112                    50.0       22.0   
526  Artic-Area                47                    84.0       49.0   
527  Artic-Area                37                    73.0       45.0   
528  Artic-Area               217                    50.0       44.0   

     colourishness  New lump inunderarm  BreastCancerLikelihood  AgeGroup  \
0              0.0                 33.6                   0.627      20.0   
1              0.0                 26.6                   0.351      41.0   
2              0.0                 23.3                   0.672      52.0   
3             94.0                 28.1                   0.167      50.0   
4            168.0                 43.1                   2.288      66.0   
..             ...                  ...                     ...       ...   
524            0.0                 31.6                   0.151      30.0   
525            0.0                 21.8                   0.444      30.0   
526           72.0                 18.2                   0.299      41.0   
527           25.0                 26.3                   0.107      54.0   
528          118.0                 30.8                   0.493      42.0   

     Results    0    1    2    3    4    5    6    7  
0          1  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  
1          0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  
2          1  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  
3          0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  
4          1  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  
..       ...  ...  ...  ...  ...  ...  ...  ...  ...  
524        0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  
525        0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  
526        0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  
527        0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  
528        0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  

[529 rows x 17 columns]

In [17]:
#The 4 preprocessing technique to be used

In [18]:
# importing other preprocessing techniques
from sklearn.preprocessing import scale,MinMaxScaler
from sklearn.impute import SimpleImputer

In [19]:
#object for the preprocessing
min_max = MinMaxScaler()

imp = SimpleImputer(missing_values=np.nan, strategy='mean')

In [20]:
# now spliting data into dependent and independent variables
X = BreastCancerPrediction_clean.loc[:,['Nipple discharge','Change of breast shap','Thickning','colourishness','New lump inunderarm','BreastCancerLikelihood','AgeGroup']]

y = BreastCancerPrediction_clean['Results']


C:\Users\MATTHEW\Anaconda3\lib\site-packages\pandas\core\indexing.py:1418: FutureWarning: 
Passing list-likes to .loc or [] with any missing label will raise
KeyError in the future, you can use .reindex() as an alternative.

See the documentation here:
https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike
  return self._getitem_tuple(key)


In [21]:
# putting the preprocessing technique on the data

In [22]:
# split dataset into train and test data
from sklearn.model_selection import train_test_split
train_X,test_X,train_y,test_y = train_test_split(X, y,train_size=0.5,test_size=0.5,random_state=123) 


In [23]:
# Now going through with logistic regression
#1. SimpleImputer
imp.fit(train_X)
imp.fit(test_X)

xnumpy = imp.transform(train_X)
xtestnum = imp.transform(test_X)

# converting numpy to dataframe
Xdf = pd.DataFrame(xnumpy)
testx = pd.DataFrame(xtestnum)

In [24]:
# importing logistic regeression model on the preprocessed data
from sklearn.linear_model import LogisticRegression

In [25]:
#fit the model
log = LogisticRegression(penalty='l2',C=.01)
log.fit(Xdf,train_y)

C:\Users\MATTHEW\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


LogisticRegression(C=0.01, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='warn', n_jobs=None, penalty='l2',
                   random_state=None, solver='warn', tol=0.0001, verbose=0,
                   warm_start=False)

In [26]:
# applying the model
log.predict(testx)

array([0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,
       1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0], dtype=int64)

In [27]:
# checking the accuracy
log.score(testx,test_y)

0.6264150943396226

# 2. MinMaxScaler


In [28]:
#using the minmax
x_train_minMax = min_max.fit_transform(xnumpy)
x_test_minMax = min_max.fit_transform(xtestnum)

In [29]:
# fitting model with minMax scaler
log2 =  LogisticRegression(penalty='l2',C=.01)
log2.fit(x_train_minMax,train_y)

C:\Users\MATTHEW\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


LogisticRegression(C=0.01, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='warn', n_jobs=None, penalty='l2',
                   random_state=None, solver='warn', tol=0.0001, verbose=0,
                   warm_start=False)

In [30]:
# Using the model
log2.predict(x_test_minMax)

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0], dtype=int64)

In [31]:
# accuracy
log.score(x_test_minMax,test_y)

0.6490566037735849

# 3 MinMax Scaler

In [32]:
x_train_scale = scale(xnumpy)
x_test_scale = scale(xtestnum)

In [33]:
# fitting model
log3 = LogisticRegression(penalty='l2',C=.01)
log3.fit(x_train_scale,train_y)

C:\Users\MATTHEW\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


LogisticRegression(C=0.01, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='warn', n_jobs=None, penalty='l2',
                   random_state=None, solver='warn', tol=0.0001, verbose=0,
                   warm_start=False)

In [34]:
# using the  model
log3.predict(x_test_scale)

array([1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       1, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0,
       0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1,
       0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 1, 1,
       0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0,
       0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0,
       1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
       0, 0, 0, 1, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0,
       0, 1, 1, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0,
       0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0,
       0], dtype=int64)

In [35]:
# checking the accuracy
log3.score(x_test_scale,test_y)

0.6867924528301886

# 4. Iterative Imputer

In [36]:
from sklearn.experimental import enable_iterative_imputer
from sklearn.impute import IterativeImputer
Itmp = IterativeImputer(max_iter=10,random_state=0)

In [37]:
Itmp.fit(train_X)
Itmp.fit(test_X)

x_train_std = Itmp.transform(train_X)
xtest_std = Itmp.transform(test_X)

# converting numpy to dataframe
x_train_std_pd = pd.DataFrame(x_train_std)
xtest_std_pd = pd.DataFrame(xtest_std)

In [38]:
# fitting model
log4 = LogisticRegression(penalty='l2',C=.01)
log4.fit(x_train_std,train_y)

C:\Users\MATTHEW\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


LogisticRegression(C=0.01, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='warn', n_jobs=None, penalty='l2',
                   random_state=None, solver='warn', tol=0.0001, verbose=0,
                   warm_start=False)

In [39]:
# using the model
log4.predict(xtest_std_pd)

array([0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,
       1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0], dtype=int64)

In [40]:
# checking the accuracy
log4.score(xtest_std_pd,test_y)

0.630188679245283

# USING THE KNN MODEL

In [41]:
from sklearn.neighbors import KNeighborsClassifier

Knn = KNeighborsClassifier(n_neighbors=5)

# 1.USING THE SIMPLE IMPUTER

In [42]:

#pip3 install tensorflow

#1. SimpleImputer
imp.fit(train_X)
imp.fit(test_X)

x_array = imp.transform(train_X)
xtest_array = imp.transform(test_X)

# converting numpy to dataframe
Xframe = pd.DataFrame(x_array)
testx = pd.DataFrame(xtest_array)

In [43]:
# fitting model
Knn.fit(Xframe,train_y)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=5, p=2,
                     weights='uniform')

In [44]:
# using the model to do the prediction
Knn.predict(testx)

array([1, 1, 1, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0,
       0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1,
       0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1,
       0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 0,
       0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1,
       0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1,
       1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1, 1,
       0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 1, 0,
       0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       1, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
       1], dtype=int64)

In [45]:
# Checking the level of accuracy of the model
Knn.score(testx,test_y)

0.6679245283018868

In [46]:
#2. MinMaxScaler
x_train_minMax = min_max.fit_transform(x_array)
x_test_minMax = min_max.fit_transform(xtest_array)

In [47]:
# fitting model with minMax scaler
Knn2 = KNeighborsClassifier(n_neighbors=5)
Knn2.fit(x_train_minMax,train_y)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=5, p=2,
                     weights='uniform')

In [48]:
# applying the model
Knn2.predict(x_test_minMax)

array([1, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 0,
       1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 1, 0,
       0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0,
       0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1,
       0, 1, 1, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 1,
       0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0,
       1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1,
       1, 1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1,
       0, 1, 1, 0, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 0,
       0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0,
       0, 0, 1, 1, 0, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 1, 0, 0,
       0], dtype=int64)

In [49]:
# checking the accuracy
Knn2.score(x_test_minMax,test_y)

0.6415094339622641

In [50]:
#3. SCALE
x_train_scale = scale(x_array)
x_test_scale = scale(xtest_array)

In [51]:
# fitting model with Scale
Knn3 = KNeighborsClassifier(n_neighbors=5)
Knn3.fit(x_train_scale,train_y)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=5, p=2,
                     weights='uniform')

In [52]:
# using the model
Knn3.predict(x_test_scale)

array([0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0,
       1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0,
       0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1,
       0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 1, 0,
       0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 1,
       0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0,
       1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 1,
       0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0,
       1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0,
       0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0,
       0], dtype=int64)

In [53]:
# checking the accuracy
Knn3.score(x_test_minMax,test_y)

0.45660377358490567

# 4. Iterative Imputer

In [54]:
Itmp.fit(train_X)
Itmp.fit(test_X)

x_train_std = Itmp.transform(train_X)
xtest_std = Itmp.transform(test_X)

# converting numpy to dataframe
x_train_std_pd = pd.DataFrame(x_train_std)
xtest_std_pd = pd.DataFrame(xtest_std)

In [55]:
# fitting model with Scale
Knn4 = KNeighborsClassifier(n_neighbors=5)
Knn4.fit(x_train_std_pd,train_y)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=5, p=2,
                     weights='uniform')

In [57]:
# applying the model
Knn4.predict(xtest_std_pd)


array([1, 1, 1, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0,
       0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1,
       0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1,
       0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 0,
       0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1,
       0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1,
       1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1, 1,
       0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 1, 0,
       0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       1, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
       1], dtype=int64)

In [58]:
# checking the accuracy
Knn4.score(testx,test_y)

0.6679245283018868

# GENERAL OBSERVATION
1. One of the observations made is that the data after, applying the logistic regressions four times gave an average accuracy of 63%, 64% and 68% accuracy
2. the KNN model also returnred a value that averagely of 62%
